modules/apps/headless/headless-admin-taxonomy/headless-admin-taxonomy-impl/src/main/java/com/liferay/headless/admin/taxonomy/internal/dto/v1_0/converter/TaxonomyCategoryDTOConverter.java

```diff
@@ -169,6 +164,50 @@ private TaxonomyCategory _toTaxonomyCategory(
 							assetCategory.getParentCategory(),
 							dtoConverterContext);
 					});
+
+				setTaxonomyCategoryUsageCount(
+					() -> {
+						Optional<UriInfo> uriInfoOptional =
+							dtoConverterContext.getUriInfoOptional();
+
+						List<String> restrictFieldsList =
+							Collections.emptyList();
+
+						if (uriInfoOptional.isPresent()) {
+							UriInfo uriInfo = uriInfoOptional.get();
+
+							MultivaluedMap<String, String> queryParameters =
+								uriInfo.getQueryParameters();
+
+							String restrictFields = queryParameters.getFirst(
+								"restrictFields");
+
+							String[] restrictFieldsArray = restrictFields.split(
```

Comment:
Maybe better to use this method from the `StringUtil` class:

`public static List<String> split(String s)`

`restictFieldsList = StringUtil.split(restrictFields);`

Reply:
In fact, I think we could do something more magical :magic_wand: to use less variables: 

`						
						Optional<UriInfo> uriInfoOptional =
							dtoConverterContext.getUriInfoOptional();

						if (uriInfoOptional.isPresent()) {
							UriInfo uriInfo = uriInfoOptional.get();

							MultivaluedMap<String, String> queryParameters =
								uriInfo.getQueryParameters();

							String restrictFields = queryParameters.getFirst(
								"restrictFields");

							if (ListUtil.exists(
									StringUtil.split(restrictFields),
									"taxonomyCategoryUsageCount"::equals)) {

								return null;
							}
						}
`

Maybe too magical :thinking:

Reply:
True! Done! Thanks Carlos 🙂

Reply:
Well, my first implementation was something like
```
List<String> restrictFields = uriInfoOptional
							.map(UriInfo::getQueryParameters)
							.map(queryParameters -> queryParameters.getFirst(
								"restrictFields"))
							.map(StringUtil::split)
							.map(Arrays::asList)
							.orElse(Collections.emptyList());
						
						if(restrictFields.contains("taxonomyCategoryUsageCount")) {
```

But the I remembered that Brian's does not like streams, so... 🤷‍♂️